### PR TITLE
Retry pending deployments longer before failing them

### PR DESCRIPTION
**Pull request by kargakis**
@mfojtik ptal

### CODE REVIEW

**Comment by kargakis** [comment]
@mfojtik didn't we see this flake somewhere? Failed deployment because of a missing deployer but the deployer actually existed?

**Comment by kargakis** [comment]
[test]

**Comment by mfojtik** [comment]
@kargakis yes, there was one. good catch!

File: pkg/deploy/controller/deployment/controller.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/openshift/origin/pkg/util"
 )
 
+const maxRetryCount = 10
```
Comment (mfojtik) [inline pkg/deploy/controller/deployment/controller.go]: please drop a comment here explaining what is this retry count

**Comment by mfojtik** [comment]
this LGTM

**Comment by kargakis** [comment]
[merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/11114 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/12072 [merge]

File: pkg/deploy/controller/deployment/controller.go [hunk 2]
```diff
@@ -20,6 +20,21 @@ import (
 	"github.com/openshift/origin/pkg/util"
 )
 
+// maxRetryCount is the maximum number of times the controller will retry errors.
+// The first requeue is after 5ms and subsequent requeues grow exponentially.
+// This effectively can extend up to 5^10ms which caps to 1000s:
+//
+// 5ms, 25ms, 125ms, 625ms, 3s, 16s, 78s, 390s, 1000s, 1000s
```
Comment (kargakis) [inline pkg/deploy/controller/deployment/controller.go]: @bparees @csrwng you guys need to be aware for what you are signing up with 60 retries ^

Comment (csrwng) [inline pkg/deploy/controller/deployment/controller.go]: thx @kargakis What determines the rate of backoff for each requeue? Is it configurable?

Comment (kargakis) [inline pkg/deploy/controller/deployment/controller.go]: Not configurable currently. See [here](https://github.com/kubernetes/kubernetes/blob/7d040dd12854795ed63115fb0d95facdc39bd60c/staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go#L41).

Comment (kargakis) [inline pkg/deploy/controller/deployment/controller.go]: I guess though you can construct your own ratelimiter

Comment (csrwng) [inline pkg/deploy/controller/deployment/controller.go]: thx

**Pull request merged**